### PR TITLE
Add flake8-logging and flake8-logging-format ruff rules

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes
Add [flake8-logging](https://docs.astral.sh/ruff/rules/#flake8-logging-log) and [flake8-logging-format](https://docs.astral.sh/ruff/rules/#flake8-logging-format-g) ruff rules.
## Issue ticket number and link
#995

### CODE REVIEW

File: griptape/drivers/vector/opensearch_vector_store_driver.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def load_entry(self, vector_id: str, *, namespace: Optional[str] = None) -> Opti
             else:
                 return None
         except Exception as e:
-            logging.error(f"Error while loading entry: {e}")
+            logging.exception("Error while loading entry: %s", e)
```
Comment (dylanholmes) [inline griptape/drivers/vector/opensearch_vector_store_driver.py:96]: I was going to complain about not using f-strings, but the argument in [G004](https://docs.astral.sh/ruff/rules/logging-f-string/) about deferring string formatting speaks to me. TIL

(Would be even cooler if the formatted values could be lazily computed too)

File: griptape/tasks/actions_subtask.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def attach_to(self, parent_task: BaseTask) -> None:
             else:
                 self.__init_from_artifacts(self.input)
         except Exception as e:
-            self.structure.logger.error(f"Subtask {self.origin_task.id}\nError parsing tool action: {e}")
+            self.structure.logger.error("Subtask %s\nError parsing tool action: %s", self.id, e)
```
Comment (dylanholmes) [inline griptape/tasks/actions_subtask.py:77]: Was `self.origin_task.id` -> `self.id` intentional?

File: griptape/tasks/base_audio_generation_task.py [hunk 3]
```diff
@@ -13,9 +13,9 @@ class BaseAudioGenerationTask(BlobArtifactFileOutputMixin, RuleMixin, BaseTask,
     def before_run(self) -> None:
         super().before_run()
 
-        self.structure.logger.info(f"{self.__class__.__name__} {self.id}\nInput: {self.input.to_text()}")
+        self.structure.logger.info("%s %s\nInput: %s", self.__class__.__name__, self.id, self.input.to_text())
```
Comment (dylanholmes) [inline griptape/tasks/base_audio_generation_task.py:16]: This is one case where I might favor readability over the micro-optimization of lazy string formatting, but 🤷

**Review by dylanholmes — COMMENTED** [review]
Just one real question

File: griptape/tasks/actions_subtask.py (hunk 2 above)
Comment (collindutter) [reply griptape/tasks/actions_subtask.py:77]: Good catch, no

File: griptape/tasks/base_audio_generation_task.py (hunk 3 above)
Comment (collindutter) [reply griptape/tasks/base_audio_generation_task.py:16]: Agreed, but 🤷

**Review by dylanholmes — APPROVED** [review]

**Pull request merged**